environments/common/layouts/everything

```diff
@@ -35,4 +35,7 @@ control
 control
 
 [filebeat:children]
-slurm_stats
\ No newline at end of file
+slurm_stats
+
+[rebuild:children]
+compute
```

Comment:
Only issue here is that the vagrant example is using this file and it will fall the validation as it is not an openstack cloud, see: https://github.com/stackhpc/openhpc-demo/blob/c9fc16510513f1a655049c7576f9ce672a850f18/environments/vagrant-example/ansible.cfg#L8.

Reply:
Now failing with this error: https://github.com/stackhpc/openhpc-demo/pull/44/checks?check_run_id=2085683879#step:7:106

Reply:
We've got some incompatible pieces here:
- Defining a group to turn on rebuild seems "right" as it's consistent with the way we do everything else
- Testing the `everything` template using vagrant seems right

Seems like there's a few approaches which would keep both of the above:
- Modify the inventory groups when running vagrant. Not easy to do directly
- Special-case the rebuild validate & deploy when running vagrant. Messy, and probably the problems above are really a specific case of a more general "we can't do this in vagrant" problem.
- Make vagrant pass the validation task - the "main" looks like it should be ok - might need `openhpc_rebuild_reconfigure: false`

Reply:
> We've got some incompatible pieces here:
> 
> * Defining a group to turn on rebuild seems "right" as it's consistent with the way we do everything else
> * Testing the `everything` template using vagrant seems right
> 
> Seems like there's a few approaches which would keep both of the above:
> 
> * Modify the inventory groups when running vagrant. Not easy to do directly
> * Special-case the rebuild validate & deploy when running vagrant. Messy, and probably the problems above are really a specific case of a more general "we can't do this in vagrant" problem.
> * Make vagrant pass the validation task - the "main" looks like it should be ok - might need `openhpc_rebuild_reconfigure: false`

To me it seems like the right place to declare an openstack cloud is in the inventory generated by the provisioning script as it has all the info about how the machines were provisioned. We should only install the openstack rebuild script on an openstack cloud. We could do this with a feature flag or via computed inventory (put the hosts in openstack_rebuild if the cloud is openstack). What do you think?

Reply:
Have implemented @jovial's proposed approach in c20faac.